File: frontend/src/types/formTemplate.ts
// Backend: /api/form-templates
import type { PageResponse } from './common'

export interface FormTemplate {
  id: number
  code: string
  category: string
  title: string
  description?: string | null
  fileName: string
  filePath: string
  fileSize: number
  version: string
  downloadCount: number
  active: boolean
  createdAt?: string
}

export interface FormTemplateListParams {
  category?: string
  keyword?: string
  page?: number
  size?: number
}

export type FormTemplatePage = PageResponse<FormTemplate>

File: backend/src/main/java/com/penocean/ehs/model/FormTemplate.java
package com.penocean.ehs.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class FormTemplate {

    private Long id;
    private String category;
    private String title;
    private String description;
    private String fileName;
    private String filePath;
    private Long fileSize;
    private String mimeType;
    private String version;
    private Long downloadCount;
    private Long uploadedBy;
    private Boolean active;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean deleted;
}
